src/main/java/jenkins/plugins/office365connector/Webhook.java

```diff
@@ -149,7 +149,10 @@ public void setTimeout(int timeout) {
     }
 
     public List<Macro> getMacros() {
-        return macros;
+		if (macros == null) {
```

Comment:
As I wrote, this problem is solved in line 150 so what is sense to have this dead code?

Can you provide production case for this problem?

Reply:
it's not dead code. the line 150 only covers when people actually saves the new webhook property. Imagine someone coming from an old version, where the loaded object's macro field is null.

Reply:
Very easy to test:

Reply:
or coming from 3.0
```
<properties>
    <jenkins.plugins.office365connector.WebhookJobProperty plugin="Office-365-Connector@3.0">
      <webhooks>
        <jenkins.plugins.office365connector.Webhook>
          <url>https://office.com</url>
          <startNotification>false</startNotification>
          <notifySuccess>true</notifySuccess>
          <notifyAborted>false</notifyAborted>
          <notifyNotBuilt>false</notifyNotBuilt>
          <notifyUnstable>true</notifyUnstable>
          <notifyFailure>true</notifyFailure>
          <notifyBackToNormal>true</notifyBackToNormal>
          <notifyRepeatedFailure>false</notifyRepeatedFailure>
          <timeout>30000</timeout>
        </jenkins.plugins.office365connector.Webhook>
      </webhooks>
    </jenkins.plugins.office365connector.WebhookJobProperty>
  </properties>
```

Then update to the current master branch, and hitting build without resaving or updating the the project.

Reply:
To actually update their XML object on plugin update would be to add an readResolve.

Reply:
This assumption
> when people actually saves the new webhook property. Someone coming from an old version

is not correct because there is no release version that supports macro. If you are using unreleased version than this is your risk to have some problem. If you meet this problem when upgrading from official version into next official, then we should maintain this issue.

You are using SNAPSHOT version which means it may be unstable.

Reply:
If you were to release now. it would still cause a null pointer.

Reply:
I'll happy write a unit test that loads old configuration to prove it.

Reply:
>is not correct because there is no release version that supports macro. If you are using unreleased version than this is your risk to have some problem. If you meet this problem when upgrading from official version into next official, then we should maintain this issue.

Reply:
It will not because of this https://github.com/jenkinsci/office-365-connector-plugin/pull/42/files#diff-4d256ee5f164b70f36b78f29a615087cR50 IMO this proves that right now if you upgrade from recent release, macro will never be null

Reply:
That is only true if you go to each job and hit save

Reply:
My bad, I was able to reproduce it with scenario when the base version was the one which is already released.

So then I believe empty collection https://github.com/jenkinsci/office-365-connector-plugin/pull/42/files#diff-4d256ee5f164b70f36b78f29a615087cL50 should be removed as it has no sense now and also setter https://github.com/jenkinsci/office-365-connector-plugin/pull/42/files#diff-4d256ee5f164b70f36b78f29a615087cR67 does not have to fix macro as this redundants code from getter

Reply:
I disagree with removing the setter fix as it is just as important to correct the XML data state. When we have the chance 👍